api-datatypes.go

```diff
@@ -104,10 +114,37 @@ type ObjectInfo struct {
 	// The class of storage used to store the object.
 	StorageClass string `json:"storageClass"`
 
+	// Versioning related information
+	IsLatest       bool
+	IsDeleteMarker bool
+	VersionID      string `xml:"VersionId"`
+
 	// Error
 	Err error `json:"-"`
 }
 
+// ObjectVersionInfo container for object version result.
+type ObjectVersionInfo struct {
```

Comment:
There is no reason to have ObjectVersionInfo() - ObjectInfo carries all that is needed for ObjectVersionInfo

Reply:
> There is no reason to have ObjectVersionInfo() - ObjectInfo carries all that is needed for ObjectVersionInfo

yeah, deleted